pages/news_item/news_item.sql

```diff
@@ -7,7 +7,8 @@ WITH mark_news_item_as_read AS (
 )
 SELECT
     ni.*,
-    format_date_only_no_tz(ni.date, coalesce(ci.display_timezone, c.display_timezone)) AS formatted_date
+    format_date_only_no_tz(ni.date, coalesce(ci.display_timezone, c.display_timezone)) AS formatted_date,
+    (users_is_course_staff($user_id) and ni.visible_to_students) AS show_student_badge
```

Comment:
This is good, but I think we should change `users_is_course_staff()` to `STABLE` (currently it's `VOLATILE`). See https://www.postgresql.org/docs/12/xfunc-volatility.html for details. I believe it should be `STABLE`, and that would prevent it from being re-called on every row of this query.